src/zarr/abc/store.py

```diff
@@ -112,49 +74,48 @@ async def _open(self) -> None:
         ------
         ValueError
             If the store is already open.
-        FileExistsError
-            If ``mode='w-'`` and the store already exists.
-
-        Notes
-        -----
-        * When ``mode='w'`` and the store already exists, it will be cleared.
         """
         if self._is_open:
             raise ValueError("store is already open")
-        if self.mode.str == "w":
-            await self.clear()
-        elif self.mode.str == "w-" and not await self.empty():
-            raise FileExistsError("Store already exists")
         self._is_open = True
 
     async def _ensure_open(self) -> None:
         """Open the store if it is not already open."""
         if not self._is_open:
             await self._open()
 
-    @abstractmethod
-    async def empty(self) -> bool:
+    async def empty(self, prefix: str = "") -> bool:
         """
-        Check if the store is empty.
+        Check if the directory is empty.
```

Comment:
are we going with object storage vocab or file system vocab? If the former, then maybe we should say something like "check if the store contains any keys with the given prefix"

Reply:
Good question. This highlights something we've been inconsistent about. I'll argue that we should use key-value store nomenclature for our current stores. Later on, it may be very useful to introduce something like a `HighLevelStore` that knows about directories, arrays, groups, etc.